pkg/receive/handler.go

```diff
@@ -430,35 +424,14 @@ func (h *Handler) handleRequest(ctx context.Context, rep uint64, tenant string,
 	return h.forward(ctx, tenant, r, wreq)
 }
 
-func (h *Handler) isTenantValid(tenant string) error {
-	if tenant != path.Base(tenant) {
-		return errors.New("Tenant name not valid")
-	}
-	return nil
-}
-
 func (h *Handler) receiveHTTP(w http.ResponseWriter, r *http.Request) {
 	var err error
 	span, ctx := tracing.StartSpan(r.Context(), "receive_http")
 	defer span.Finish()
 
-	tenant := r.Header.Get(h.options.TenantHeader)
-	if tenant == "" {
-		tenant = h.options.DefaultTenantID
-	}
-
-	if h.options.TenantField != "" {
-		tenant, err = h.getTenantFromCertificate(r)
-		if err != nil {
-			// This must hard fail to ensure hard tenancy when feature is enabled.
-			http.Error(w, err.Error(), http.StatusBadRequest)
-			return
-		}
-	}
-
-	err = h.isTenantValid(tenant)
+	tenant, err := tenancy.GetTenantFromHTTP(r, h.options.TenantHeader, h.options.DefaultTenantID, h.options.TenantField)
 	if err != nil {
-		level.Error(h.logger).Log("msg", "tenant name not valid", "tenant", tenant)
+		level.Error(h.logger).Log("err", err)
```

Comment:
```suggestion
		level.Error(h.logger).Log("error getting tenant from HTTP", err)
```

Reply:
Are we OK with having a full sentence to the first argument there? Usually in the rest of the codebase it's single words, most commonly `Log("msg",...` , seen also `err` or `warnings`, never saw full sentences like that.

Reply:
You're right about the pattern. Then how about  `Log(msg, "getting tenant from HTTP", "err", err)`. I'd like this log to be a bit descriptive, as an implication of not being able to get the tenant is bad here! 

We have similar full sentence logs like `Log("msg", "getting tracing config failed", "err", err)` elsewhere too.